FWCore/Framework/src/InputSource.cc

```diff
@@ -42,6 +44,14 @@ namespace edm {
         return th;
       return (lastDigit == 1 ? st : (lastDigit == 2 ? nd : rd));
     }
+
+    std::string guid() {
+      Service<ProcessGUID> svc;
+      if (svc.isAvailable()) {
+        return svc->binary();
+      }
+      return createGlobalIdentifier(true);
+    }
```

Comment:
I tried first to always use `Service<ProcessGUID>()->binary()`, but many of our non-`cmsRun` tests use `Source` outside of `cmsRun`, and I wasn't sure if it would be better to migrate all of them (and any other non-`cmsRun` use case) or fall back to creating a GUID directly.

Reply:
Which tests?

Reply:
E.g. here
https://github.com/cms-sw/cmssw/blob/1d3499ad1b75471d95460bfe1d676295dbe88842/FWCore/Framework/test/eventprocessor_t.cppunit.cc#L79
https://github.com/cms-sw/cmssw/blob/1d3499ad1b75471d95460bfe1d676295dbe88842/FWCore/Framework/test/eventprocessor_t.cppunit.cc#L116
https://github.com/cms-sw/cmssw/blob/1d3499ad1b75471d95460bfe1d676295dbe88842/FWCore/Framework/test/productregistry.cppunit.cc#L232
https://github.com/cms-sw/cmssw/blob/1d3499ad1b75471d95460bfe1d676295dbe88842/FWCore/Integration/test/standalone_t.cppunit.cc#L59
(after which I stopped looking)

> Hmmm. Since this is not configurable, maybe just a singleton would be better? We do that for the ParameterSet registry.

Would that make it easier to have it "always available"? Could be worth a try.

Reply:
> Would that make it easier to have it "always available"? Could be worth a try.

It would avoid the Service dependency which, now that I see the code, doesn't seem to be adding anything overly useful.